mempool/src/shared_mempool.rs

```diff
@@ -297,15 +297,12 @@ where
     )
     .await;
 
-    // eagerly filter out transactions that were already committed
     let transactions: Vec<_> = transactions
         .into_iter()
         .enumerate()
         .filter_map(|(idx, t)| {
             if let Ok((sequence_number, balance)) = account_states[idx] {
-                if t.sequence_number() >= sequence_number {
-                    return Some((t, sequence_number, balance));
-                }
+                return Some((t, sequence_number, balance));
```

Comment:
what's a point of having this filter at all now?
it's like extra I/O without any real check